addon/-private/system/model/internal-model.js

```diff
@@ -235,8 +241,68 @@ InternalModel.prototype = {
     return this.record;
   },
 
-  unloadRecord() {
+  directlyRelatedInternalModels() {
+    var array = [];
+    this.type.eachRelationship((key, relationship) => {
```

Comment:
we may want to use `eachRelationshipByName` cp here: https://github.com/emberjs/data/blob/c9d8212c857ca78218ad98d11621819b38dba98f/addon/-private/system/relationships/ext.js#L486 because it is cached (eachRelBy Name can be costly), and seems like its provides what we want here.